File: sdk/ruby/lib/rocksky/resources/spotify.rb
module Rocksky
  module Resources
    # `app.rocksky.spotify.*` endpoints — Spotify remote control for the
    # authenticated user.
    class Spotify < Base
      # Currently playing on Spotify.
      def get_currently_playing(actor: nil)
        query("app.rocksky.spotify.getCurrentlyPlaying", actor: actor)
      end

      # Resume Spotify playback.
      def play
        procedure("app.rocksky.spotify.play")
      end

      # Pause Spotify playback.
      def pause
        procedure("app.rocksky.spotify.pause")
      end

      # Skip to next track.
      def next
        procedure("app.rocksky.spotify.next")
      end

      # Go to previous track.
      def previous
        procedure("app.rocksky.spotify.previous")
      end

      # Seek to position in milliseconds.
      def seek(position:)
        procedure("app.rocksky.spotify.seek", params: { position: position })
      end
    end
  end
end

File: sdk/python/src/rocksky/resources/spotify.py
"""``app.rocksky.spotify.*`` — Spotify playback control proxy."""

from __future__ import annotations

from typing import Any

from ._base import Resource


class SpotifyResource(Resource):
    async def currently_playing(self, *, actor: str | None = None) -> Any:
        return await self._transport.query(
            "app.rocksky.spotify.getCurrentlyPlaying",
            params={"actor": actor},
        )

    async def play(self) -> Any:
        return await self._transport.procedure("app.rocksky.spotify.play", auth=True)

    async def pause(self) -> Any:
        return await self._transport.procedure("app.rocksky.spotify.pause", auth=True)

    async def next(self) -> Any:
        return await self._transport.procedure("app.rocksky.spotify.next", auth=True)

    async def previous(self) -> Any:
        return await self._transport.procedure("app.rocksky.spotify.previous", auth=True)

    async def seek(self, position: int) -> Any:
        return await self._transport.procedure(
            "app.rocksky.spotify.seek", params={"position": position}, auth=True
        )
